debezium-server-iceberg-sink/src/main/java/io/debezium/server/iceberg/tableoperator/IcebergTableOperator.java

```diff
@@ -60,13 +61,33 @@ protected List<EventConverter> deduplicateBatch(List<EventConverter> events) {
 
           try {
             // deduplicate using key(PK)
-            deduplicatedEvents.merge(e.key(), e, (oldValue, newValue) -> {
-              if (this.compareByTsThenOp(oldValue, newValue) <= 0) {
-                return newValue;
-              } else {
-                return oldValue;
+            if (cdcKeepLatest) {
+              EventConverter prev = deduplicatedEvents.get(e.key());
+              if (prev != null) {
+                Long prevTs = prev.cdcSourceTsValue();
+                Long ts = e.cdcSourceTsValue();
+                Operation prevOp = prev.cdcOpValue();
+                Operation op = e.cdcOpValue();
+                if (Long.compare(prevTs, ts) > 0 ||
+                    (prevOp == Operation.INSERT && op == Operation.INSERT) ||
+                    (prevOp == Operation.UPDATE && op == Operation.INSERT) ||
+                    (prevOp == Operation.DELETE && op == Operation.UPDATE) ||
+                    (prevOp == Operation.DELETE && op == Operation.DELETE)
+                ) {
+                  LOGGER.warn("Out of order events for key {} detected: {} {} -> {} {}", e.key(), prevTs, prevOp, ts, op);
```

Comment:
could we create a test to trigger this situation?  IcebergTableOperatorTest.testDeduplicateBatch has a way to create manual events and de-duplicate them.